pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -571,6 +571,17 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *v1.PersistentVolume)
 			}
 			return nil
 		} else if claim.Spec.VolumeName == "" {
+			if isMisMatch, err := checkVolumeModeMisMatches(&claim.Spec, &volume.Spec); err != nil || isMisMatch {
+				// Binding for the dynamically provisioned volume won't be called in syncUnboundClaim,
+				// because findBestMatchForClaim won't return the volume due to volumeMode mismatch.
+				volumeMsg := fmt.Sprintf("Dynamically provisioned volume[%s] has incompatible volumeMode. This provisioner may not support the requested volumeMode.", volume.Name)
```

Comment:
At this point, it may not actually be dynamic provisioning.  The user could have manually created a PV with ClaimRef set.  So maybe the error message needs to be changed to just say "Cannot bind PersistentVolume to requested PersistentVolumeClaim due to incompatible volumeMode."

Reply:
Thank you for pointing it out. I will fix it as suggested here and below.